src/style-spec/function/definitions/match.js

```diff
@@ -107,15 +107,12 @@ class Match implements Expression {
         for (let i = 0; i < labels.length; i++) {
             if (i > 0) lookup += ', ';
             const label = labels[i];
-            lookup += `${String(label)}: ${outputs[this.cases[label]]}`;
+            lookup += `${JSON.stringify(label)}: ${outputs[this.cases[label]]}`;
```

Comment:
Why `JSON.stringify` rather than `String`? The input is going to implicitly go through `String` in `${lookupObject}[${input}]`, so if they behave differently that would seem to be an issue.

Reply:
The `JSON.stringify()` here is to get a JS string literal for the key in the JS object literal we're building.  If not, then `label === "foo-bar"` would result in invalid code `{ foo-bar: output }`

Reply:
Looks like `JSON.stringify(number)` is equivalent to `String(number)`, so that should be fine. For strings, [U+2028 and U+2029 need special handling](http://timelessrepo.com/json-isnt-a-javascript-subset).

Reply:
Oy.  We'll need to handle those in `Literal`, too, then.